thorntail-runner/src/main/java/org/wildfly/swarm/runner/WarBuilder.java

```diff
@@ -145,8 +145,15 @@ private void addWebappResourceToWar(Path path) {
         File file = path.toFile();
         if (file.isFile()) {
             try {
-                String projectDir = Paths.get("src", "main", "webapp").toFile().getAbsolutePath();
-
+                String projectDir = System.getProperty("thorntail.runner.webapp-location");
+                if (projectDir != null) {
+                    projectDir = Paths.get(projectDir).toFile().getAbsolutePath();
+                    if (file.getAbsolutePath().contains("WEB-INF" + File.separator + "classes")) { // Ignore classes.
+                        return;
+                    }
+                } else {
+                    projectDir = Paths.get("src", "main", "webapp").toFile().getAbsolutePath();
+                }
```

Comment:
I took a look at the code and found this is really 2 different things:

1. The `addWebappResourceToWar` method doesn't respect the `thorntail.runner.webapp-location` setting. The `addWebAppResourcesToWar` method should really pass the `webappPath` to `addWebappResourceToWar` when invoking it. The `addWebappResourceToWar` method shouldn't reread the system property all the time. This should be a simple change.

2. Filtering out `WEB-INF/classes`. I think the `thorntail.runner.webapp-location` was meant to point to a _source_ directory. Pointing it to a _target_ directory requires such filtering, but I'm not sure if just ignoring `WEB-INF/classes` is enough. How about `WEB-INF/lib`? That should be enough, probably.

I think we probably can filter out these 2 paths unconditionally, even if the system property points to a source directory -- the source directory shouldn't contain these directories at all.

Reply:
I can make this change by ignoring the lib directory as well. Soon I will send.

Thank you very much for feedback.